jbpm-services/jbpm-kie-services/src/test/java/org/jbpm/kie/services/test/RuntimeDataServiceImplTest.java

```diff
@@ -446,6 +446,38 @@ public void testGetProcessInstancesByDeploymentIdNull() {
         runtimeDataService.getProcessInstancesByDeploymentId(null, Collections.<Integer>emptyList(), new QueryContext());
     }
 
+	@Test(expected = NullPointerException.class)
+	public void testCountProcessInstancesByDeploymentIdNull() {
+		runtimeDataService.countProcessInstancesByDeploymentId(null, Collections.<Integer>emptyList());
+	}
+
+
+	@Test
+	public void testCountProcessInstancesByDeploymentIdAndState() {
+		assertEquals(Long.valueOf(0), runtimeDataService.countProcessInstancesByDeploymentId(deploymentUnit.getIdentifier(), Collections.emptyList()));
+		
+		List<Long> pids = new ArrayList<>();
+		int total = 10;
+		for(int i = 0; i < total; i++) {
+			pids.add(processService.startProcess(deploymentUnit.getIdentifier(), "org.jbpm.writedocument"));
+		}
+		
+		assertEquals(Long.valueOf(0), runtimeDataService.countProcessInstancesByDeploymentId(deploymentUnit.getIdentifier(), null));
+		assertEquals(Long.valueOf(0), runtimeDataService.countProcessInstancesByDeploymentId(deploymentUnit.getIdentifier(), Collections.emptyList()));
+		assertEquals(Long.valueOf(total), runtimeDataService.countProcessInstancesByDeploymentId(deploymentUnit.getIdentifier(), Arrays.asList(ProcessInstance.STATE_ACTIVE)));
+		assertEquals(Long.valueOf(0), runtimeDataService.countProcessInstancesByDeploymentId(deploymentUnit.getIdentifier(), Arrays.asList(ProcessInstance.STATE_ABORTED)));
+		assertEquals(Long.valueOf(total), runtimeDataService.countProcessInstancesByDeploymentId(deploymentUnit.getIdentifier(), Arrays.asList(ProcessInstance.STATE_ACTIVE, ProcessInstance.STATE_ABORTED)));
+
+		int suspended = 5;
+		for(int i = 0; i < suspended; i++) {
+			processService.abortProcessInstance(pids.get(i));
+		}
+		
+		assertEquals(Long.valueOf(total - suspended), runtimeDataService.countProcessInstancesByDeploymentId(deploymentUnit.getIdentifier(), Arrays.asList(ProcessInstance.STATE_ACTIVE)));
+		assertEquals(Long.valueOf(suspended), runtimeDataService.countProcessInstancesByDeploymentId(deploymentUnit.getIdentifier(), Arrays.asList(ProcessInstance.STATE_ABORTED)));
+		assertEquals(Long.valueOf(total), runtimeDataService.countProcessInstancesByDeploymentId(deploymentUnit.getIdentifier(), Arrays.asList(ProcessInstance.STATE_ACTIVE, ProcessInstance.STATE_ABORTED)));
+	}
```

Comment:
Don't forget to abort the rest of the started process instances at the end of this method, otherwise, in the after method, it won't be able to undeploy them and you will get faulty tests